dtfs-central-api/src/v1/controllers/utilisation-report-service/fee-record-correction/post-fee-record-correction.controller/index.test.ts

```diff
@@ -0,0 +1,212 @@
+jest.mock('../../../../../repositories/fee-record-correction-transient-form-data-repo');
+
+console.error = jest.fn();
+
+describe('post-fee-record-correction.controller', () => {
+  describe('postFeeRecordCorrection', () => {
+    const reportId = 1;
+    const feeRecordId = 1;
+
+    const mockFindByUserIdAndFeeRecordId = jest.fn();
+    const mockEntityManager = {} as unknown as EntityManager;
+    const mockHandleEvent = jest.fn();
+    const mockForFeeRecordIdAndReportId = jest.fn();
+
+    beforeEach(() => {
+      jest.mocked(executeWithSqlTransaction).mockImplementation(async (functionToExecute) => {
+        await functionToExecute(mockEntityManager);
+      });
+      mockForFeeRecordIdAndReportId.mockReturnValue({
+        handleEvent: mockHandleEvent,
+      });
+      FeeRecordStateMachine.forFeeRecordIdAndReportId = mockForFeeRecordIdAndReportId;
+
+      jest.spyOn(FeeRecordCorrectionTransientFormDataRepo, 'withTransaction').mockReturnValue({ findByUserIdAndFeeRecordId: mockFindByUserIdAndFeeRecordId });
+    });
+
+    afterEach(() => {
+      jest.resetAllMocks();
+    });
+
+    it('should request record correction', async () => {
+      // Arrange
+      const userId = new ObjectId().toString();
+      const user = {
+        ...aTfmSessionUser(),
+        _id: userId,
+        firstName: 'Jane',
+        lastName: 'Smith',
+      };
+      const reasons = [RECORD_CORRECTION_REASON.FACILITY_ID_INCORRECT];
+      const additionalInfo = 'here is my additional information';
+
+      const req = httpMocks.createRequest<PostFeeRecordCorrectionRequest>({
+        params: { reportId, feeRecordId },
+        body: { user },
+      });
+      const res = httpMocks.createResponse();
+
+      mockFindByUserIdAndFeeRecordId.mockResolvedValue(
+        new FeeRecordCorrectionTransientFormDataEntityMockBuilder()
+          .withUserId(userId)
+          .withFeeRecordId(feeRecordId)
+          .withFormData({ reasons, additionalInfo })
+          .build(),
+      );
+
+      // Act
+      await postFeeRecordCorrection(req, res);
+
+      // Assert
+      expect(mockFindByUserIdAndFeeRecordId).toHaveBeenCalledWith(userId, feeRecordId);
+      expect(mockForFeeRecordIdAndReportId).toHaveBeenCalledWith(feeRecordId, reportId);
+      expect(mockHandleEvent).toHaveBeenCalledTimes(1);
+      expect(mockHandleEvent).toHaveBeenCalledWith({
+        type: FEE_RECORD_EVENT_TYPE.CORRECTION_REQUESTED,
+        payload: {
+          transactionEntityManager: mockEntityManager,
+          requestedByUser: {
+            id: userId,
+            firstName: user.firstName,
+            lastName: user.lastName,
+          },
+          reasons,
+          additionalInfo,
+          requestSource: {
+            platform: REQUEST_PLATFORM_TYPE.TFM,
+            userId,
+          },
+        },
+      });
+    });
+
+    it("responds with a '200' if the record correction request is successful", async () => {
```

Comment:
should... and also could we use ${HttpStatusCode.Ok} in the it please.  should.. in the rest of the file too please